pkg/aggregator/demultiplexer_agent.go

```diff
@@ -498,6 +504,14 @@ func (d *AgentDemultiplexer) GetEventsAndServiceChecksChannels() (chan []*metric
 // AddLateMetrics buffers a bunch of late metrics. This data will be directly
 // transmitted "as-is" (i.e. no sampling) to the serializer when a flush is triggered.
 func (d *AgentDemultiplexer) AddLateMetrics(samples metrics.MetricSampleBatch) {
+	// safe-guard: if for some reasons we are receiving some metrics here despite
```

Comment:
The reason is that if a previous version of the Agent is receiving a metric with a timestamp, it will just ignore this timestamp field and process the metric as usual.
I think it's better to keep this behavior when this field is supported but the feature is disabled.